src/main/java/decodes/hdb/HdbSiteDAO.java

```diff
@@ -102,7 +103,7 @@ public HdbSiteDAO(DatabaseConnectionOwner tsdb)
 			}
 		}
 		
-		if (myDbSiteCode == null)
+		if (!siteCodeInit)
```

Comment:
AtomicBoolean *might* be better here, but given the synchronized block it's probably fine like this.

Though I'd move the synchronized to either siteCodeInit or at least before siteCodeInit; there's a small chance of a race condition or sequencing issue *if* this gets called again.

Reply:
If you're good as is, I'll merge. I would like to know more about how a race condition could be an issue here. Perhaps a deep discussion for September.

Reply:
Hard to say really, it's just that your checking the state of a variable and then synchronizing on a different object after which you set the value of the first. So not really a race condition but if two threads were calling that the both would check and start, though presumably it would get stopped at the synchronization block so maybe not an issue right now.